main/util/src/mill/util/CoursierSupport.scala

```diff
@@ -164,6 +167,47 @@ trait CoursierSupport {
     (deps0, res.getOrThrow)
   }
 
+  def jvmIndex(
+      ctx: Option[mill.api.Ctx.Log] = None,
+      coursierCacheCustomizer: Option[FileCache[Task] => FileCache[Task]] = None
+  ): Result[JvmIndex] = {
+    val coursierCache0 = coursierCache(ctx, coursierCacheCustomizer)
+    JvmIndex.load().unsafeRun()(coursierCache0.ec)
```

Comment:
You should use a `JvmIndex.load` [overload](https://github.com/coursier/coursier/blob/f1840c21352aeeee3b3008d4a728156895d33196/modules/jvm/src/main/scala/coursier/jvm/JvmIndex.scala#L183-L187) that accepts a coursier cache instance (and pass it `coursierCache0`). Note that [I'm](https://github.com/coursier/coursier/pull/3160) is going to undeprecate the linked overload, it's fine to use it.